src/benchmarks/real-world/Microsoft.ML.Benchmarks/HashBench.cs

```diff
@@ -20,73 +21,93 @@ namespace Microsoft.ML.Benchmarks
+            /// <param name="column"> is the index of a output column whose getter should be returned.</param>
+            public override ValueGetter<TValue> GetGetter<TValue>(DataViewSchema.Column column)
             {
-                if (col != 0)
+                if (column.Index != 0)
                     throw new Exception();
                 if (_getter is ValueGetter<TValue> typedGetter)
                     return typedGetter;
                 throw new Exception();
             }
 
-            public static RowImpl Create<T>(ColumnType type, ValueGetter<T> getter)
+            public static RowImpl Create<T>(DataViewType type, ValueGetter<T> getter)
             {
                 if (type.RawType != typeof(T))
                     throw new Exception();
                 return new RowImpl(type, getter);
             }
 
-            private RowImpl(ColumnType type, Delegate getter)
+            private RowImpl(DataViewType type, Delegate getter)
             {
-                var builder = new SchemaBuilder();
+                var builder = new DataViewSchema.Builder();
                 builder.AddColumn("Foo", type, null);
-                Schema = builder.GetSchema();
+                Schema = builder.ToSchema();
                 _getter = getter;
             }
         }
 
         private const int Count = 100_000;
 
-        private readonly IHostEnvironment _env = new MLContext();
+        private readonly MLContext _env = new MLContext();
 
         private RowImpl _inRow;
         private ValueGetter<uint> _getter;
         private ValueGetter<VBuffer<uint>> _vecGetter;
 
-        private void InitMap<T>(T val, ColumnType type, int hashBits = 20, ValueGetter<T> getter = null)
+        private void InitMap<T>(T val, DataViewType type, int numberOfBits = 20, ValueGetter<T> getter = null)
         {
             if (getter == null)
                 getter = (ref T dst) => dst = val;
             _inRow = RowImpl.Create(type, getter);
             // One million features is a nice, typical number.
-            var info = new HashingTransformer.ColumnInfo("Bar", "Foo", hashBits: hashBits);
-            var xf = new HashingTransformer(_env, new[] { info });
-            var mapper = xf.GetRowToRowMapper(_inRow.Schema);
+            var info = _env.Transforms.Conversion.Hash("Bar", "Foo", numberOfBits: numberOfBits);
+            var columns = typeof(HashingEstimator)
+                .GetFields(BindingFlags.NonPublic | BindingFlags.Instance)
+                .Single(field => field.Name == "_columns")
+                .GetValue(info);
+            var xf = typeof(HashingTransformer)
+                .GetConstructor(
+                    BindingFlags.NonPublic | BindingFlags.Instance, 
+                    null, 
+                    new[] { typeof(IHostEnvironment), columns.GetType() }, 
+                    Array.Empty<ParameterModifier>())
+                .Invoke(new object[] { _env, columns });
```

Comment:
I really don't like reflection, but I could not find any other way to get it working without introducing changes to ML.NET.

I was able to compile following code:

```cs
var info = _env.Transforms.Conversion.Hash("Bar", "Foo", numberOfBits: numberOfBits);
var dataView = _env.Data.LoadFromEnumerable(Array.Empty<RowImpl>(), _inRow.Schema);
var xf = info.Fit(dataView);
```

but I was getting errors like this:

```log
 System.ArgumentOutOfRangeException: Could not determine an IDataView type for member Schema
Parameter name: rawType
   at Microsoft.ML.Data.InternalSchemaDefinition.GetVectorAndItemType(String name, Type rawType, IEnumerable`1 attributes, Boolean& isVector, Type& itemType)
   at Microsoft.ML.Data.InternalSchemaDefinition.GetVectorAndItemType(MemberInfo memberInfo, Boolean& isVector, Type& itemType)
   at Microsoft.ML.Data.SchemaDefinition.Create(Type userType, Direction direction)
   at Microsoft.ML.Data.DataViewConstructionUtils.GetSchemaDefinition[TRow](IHostEnvironment env, DataViewSchema schema)
   at Microsoft.ML.Data.DataViewConstructionUtils.GetInternalSchemaDefinition[TRow](IHostEnvironment env, DataViewSchema schema)
   at Microsoft.ML.Data.DataViewConstructionUtils.CreateFromEnumerable[TRow](IHostEnvironment env, IEnumerable`1 data, DataViewSchema schema)
   at Microsoft.ML.Benchmarks.HashBench.InitMap[T](T val, DataViewType type, Int32 numberOfBits, ValueGetter`1 getter)
   at Microsoft.ML.Benchmarks.HashBench.SetupHashScalarString() in C:\Projects\performance\src\benchmarks\real-world\Microsoft.ML.Benchmarks\HashBench.cs:line 136
```

Reply:
You can remove this by cherry-picking this change: [https://github.com/eerhardt/performance/commit/db30f57f91175610b3bfbfd30e788b245f03facd](https://github.com/eerhardt/performance/commit/db30f57f91175610b3bfbfd30e788b245f03facd)

---
In reply to: [341010904](https://github.com/dotnet/performance/pull/987#discussion_r341010904) [](ancestors = 341010904)

Reply:
I've pushed the change to your branch.